Plutarch/Lift.hs

```diff
@@ -105,24 +105,26 @@ data LiftError
   | LiftError_CompilationError Text
   deriving stock (Eq)
 
-{- | Convert a Plutarch term to the associated Haskell value. Fail otherwise.
-This will fully evaluate the arbitrary closed expression, and convert the resulting value.
+{- | Convert a Plutarch term to the associated Haskell value and trace from evaluation. Fail otherwise.
+     This will fully evaluate the arbitrary closed expression, and convert the resulting value.
+
+ @since 1.2.1
 -}
-plift' :: forall p. PUnsafeLiftDecl p => Config -> ClosedTerm p -> Either LiftError (PLifted p)
-plift' config prog = case compile config prog of
-  Left msg -> Left $ LiftError_CompilationError msg
+pliftTrace :: forall p. PUnsafeLiftDecl p => Config -> ClosedTerm p -> (Either LiftError (PLifted p), [Text])
+pliftTrace config prog = case compile config prog of
+  Left msg -> (Left $ LiftError_CompilationError msg, mempty)
   Right script -> case evalScriptHuge script of
-    (Right (Scripts.unScript -> UPLC.Program _ _ term), _, _) ->
+    (Right (Scripts.unScript -> UPLC.Program _ _ term), _, trace) ->
       case readKnownConstant term of
```

Comment:
I think what we really want is something like this
```haskell
compile'' :: Config -> ClosedTerm a -> CompiledTerm a
evalCompiledTerm :: CompiledTerm a -> (Either EvalError (CompiledTerm a), ExBudget, [Text])
pliftCompiled :: CompiledTerm a -> Either KnownTypeError (Maybe (PLifted a))
```
The question is, what should the name of `compile''` be?

Reply:
I think the idea of `CompiledTerm` is very helpful. It opens the door for some big optimization. For example, it will allow shrinkers to make reasonable alternates and etc.